### DISCUSSION TITLE
Outstanding design needs for feedback mechanism

### DISCUSSION THREAD

**Discussion by vossisboss** [wagtail/guide#38] [General]
I caught up with Ben about any outstanding conversations he had with Dex about the Wagtail editor's guide, and he told me that the only really big piece left was the design for the feedback mechanism. He needs some more context from us about whether we intend to build something custom or use a third-party package to support it.

It looks like @Hitansh-Shah's original proposal was to include a smiling face/frowning face design. I like that approach and feel like that would probably give us a better idea how users are feeling about a page than a star system. I also like @Hitansh-Shah's idea to gather the feedback in a specific section of the admin area since that would make it easier for the person checking the feedback to review it all in one place (versus lots of emails).

![Screen Shot 2022-08-08 at 11 55 53 AM](https://user-images.githubusercontent.com/45881480/183461073-e71f2cb8-c268-4570-b5f2-944928990531.png)

## Some outstanding questions for discussion:

1) Who will ultimately be looking at this feedback on a regular basis? (Most likely me, but who else?)
2) Do we want to gather additional information beyond the smiling / frowning face reaction? If so, what information would be the most useful to help us improve the page?
3) Are there any well-supported third-party feedback packages available? Or would a custom build for the feedback mechanism make more sense?

My two cents on the second question is that what would be most useful to know is why someone clicked a frowning face. If they clicked the frowning face, it would be great for the user to be offer a chance to tell us what they didn't like so we can fix it or improve it. I have not turned up any appealing third-party options in my Googling so far, but definitely want to know @Hitansh-Shah's opinions on the third question.

Please weigh in there are any thoughts on these questions you can't contain. 🙂

**Comment by lb-** [comment]
I'd like us to strongly consider the built in contrib form builder for this, here is why.

- it reinforces Wagtail as a great solution for a CMS with its own features
- less maintenance
- less issues with a third party requirement for payment/coordination
- already has a built in system for email
- documentation provides a robust system for additional behaviour when submitted

I'm happy to add some code examples of a base form page and submissions model that means we don't have to do migrations later if we want to do more complex things down the road. 

As for 'where will this go' - again - built in means it's visible inside Wagtail easily. 

However, it might be possible to ask the user if they are ok to have their feedback public - and if so we could post to a GitHub discussion or raise an issue on this repo. We would need some basic moderation step for this though. It could also post to a closed Slack group pretty easily.

**Comment by allcaps** [comment]
We should keep the feedback form as simple as possible. This way it is easy to use and easy to build and maintain.
Note that the 'next level contributor' is getting access to the admin and can submit drafts for moderators approval.
